src/cli/cli.cc

```diff
@@ -1,15 +1,109 @@
 #include "cli.h"
 
+#include <unordered_map>
+
+#include "aktualizr-lite/api.h"
+
+#include "logging/logging.h"
+
 namespace cli {
 
+static StatusCode downloadRes2StatusCode(DownloadResult::Status ds);
+static StatusCode installRes2StatusCode(InstallResult::Status is);
+
 StatusCode Install(AkliteClient &client, int version) {
-  StatusCode sc{StatusCode::UnknownError};
-  return sc;
+  // Check if a device is in a correct state to start a new update
+  if (client.IsInstallationInProgress()) {
+    LOG_ERROR << "Cannot start Target installation since there is ongoing installation; target: "
+              << client.GetPendingTarget().Name();
+    return StatusCode::InstallationInProgress;
+  }
+
+  const auto current{client.GetCurrent()};
+  const CheckInResult get_tuf_res{client.CheckIn()};
+  if (get_tuf_res.status == CheckInResult::Status::Failed) {
+    LOG_ERROR << "Failed to pull TUF metadata or they are invalid";
+    return StatusCode::TufMetaPullFailure;
+  }
+
+  // Find TufTarget of the specified version, TODO: consider adding CheckInResult::GetTarget() method
```

Comment:
it already exposes a vector of Targets the user can look through. I don't mind introducing friction in an API when the caller is doing something we don't like (not installing the latest).